### ISSUE TITLE
With 1.9.0-beta4, sysimages aren't built for the specified `JULIA_CPU_TARGET`

### ISSUE THREAD

**Issue by ven-k** [JuliaLang/PackageCompiler.jl#790]
Src OS: Linux (WSL and native), CPU: znver3
Target OS: Linux, CPU: znver2
Julia: v1.9.0-beta4
PackageCompiler: 2.1.5 (throws error) and 2.1.2 (relocates, but it doesn't mention pkgimages flag at all)

----

While building the sysimage with 1.9.0-beta4, even with `cpu_target = "generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)"`, the sysimage builds only for the "native" target.
So, when this sysimage is relocated to another machine, it fails to start with `ERROR: Unable to find compatible target in system image.` error

When I printed out the cmd returned by addenv, it indicated setting of JULIA_CPU_TARGET var.
This however, isn't an issue when `--pkgimages` flag isn't mentioned and `--cpu-target` is passed explicitly, i.e. what v2.1.2 does.

With
<details>

<summary>sysimage-metadata.patch (by @staticfloat)</summary>

```
diff --git a/src/processor.cpp b/src/processor.cpp
index 13b40ec4f7..accbff5469 100644
--- a/src/processor.cpp
+++ b/src/processor.cpp
@@ -770,12 +770,16 @@ static inline SysimgMatch match_sysimg_targets(S &&sysimg, T &&target, F &&max_v
     SysimgMatch match;
     bool match_name = false;
     int feature_size = 0;
+    jl_printf(JL_STDERR, "target.name: %s\n", target.name.c_str());
+    jl_printf(JL_STDERR, "sysimg.size(): %d\n", sysimg.size());
     for (uint32_t i = 0; i < sysimg.size(); i++) {
         auto &imgt = sysimg[i];
+        jl_printf(JL_STDERR, "[%d]: %s\n", i, imgt.name.c_str());
         if (!(imgt.en.features & target.dis.features).empty()) {
             // Check sysimg enabled features against runtime disabled features
             // This is valid (and all what we can do)
             // even if one or both of the targets are unknown.
+            jl_printf(JL_STDERR, "continue 1\n");
             continue;
         }
         if (imgt.name == target.name) {
@@ -786,6 +790,7 @@ static inline SysimgMatch match_sysimg_targets(S &&sysimg, T &&target, F &&max_v
             }
         }
         else if (match_name) {
+            jl_printf(JL_STDERR, "continue 2 (match_name: %d)\n", match_name);
             continue;
         }
         int new_vsz = max_vector_size(imgt.en.features);

```
 
</details> 
to print out the metadata and
<p>
<details>

<summary>cmdline.patch</summary>

```
diff --git a/src/processor.cpp b/src/processor.cpp
index 13b40ec4f7..9b6c850e65 100644
--- a/src/processor.cpp
+++ b/src/processor.cpp
@@ -734,6 +734,12 @@ static inline jl_image_fptrs_t parse_sysimg(void *hdl, F &&callback)
 template<typename T>
 static inline void check_cmdline(T &&cmdline, bool imaging)
 {
+    jl_printf(JL_STDERR, "Imaging:, %d\n", imaging);
+    jl_printf(JL_STDERR, "cmdline size, %d\n", cmdline.size());
+    for(uint32_t i = 0; i < cmdline.size(); i++) {
+        jl_printf(JL_STDERR,"CMDLINE: [%d]: Flags: %d, Name: %s\n", i, cmdline[i].en.flags, cmdline[i].name.c_str());
+    }
+
     assert(cmdline.size() > 0);
     // It's unclear what does specifying multiple target when not generating
     // sysimg means. Make it an error for now.
```
</details>  to print out the cmdline options, I could see that it indeed ignores the cpu_target in >2.1.2 versions.

<p>

The outputs for both versions of PackageCompiler:
<details>
<summary>The output for PackageCompiler@2.1.2</summary>

```julia
julia> cpu_target = "generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)"
julia> PackageCompiler.create_sysimage(; sysimage_path="test_212.so", cpu_target)
target.name: znver3
sysimg.size(): 3
[0]: generic
[1]: sandybridge
[2]: haswell
Imaging:, 0
cmdline size, 1
CMDLINE: [0]: Flags: 0, Name: native
target.name: znver3  -----,
sysimg.size(): 1          |--- These 3 lines repeat 8 more times.
[0]: znver3          -----'
target.name: znver3
sysimg.size(): 3
[0]: generic
[1]: sandybridge
[2]: haswell
Imaging:, 0
cmdline size, 1
CMDLINE: [0]: Flags: 0, Name: native
target.name: generic
sysimg.size(): 3
[0]: generic
[1]: sandybridge
continue 1
[2]: haswell
continue 1
Imaging:, 1
cmdline size, 3
CMDLINE: [0]: Flags: 0, Name: generic
CMDLINE: [1]: Flags: 2, Name: sandybridge
CMDLINE: [2]: Flags: 0, Name: haswell
✔ [02m:39s] PackageCompiler: compiling incremental system image
```
</details>

<details>

<summary>The output for PackageCompiler@2.1.5</summary>

```julia
julia> cpu_target = "generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)"
julia> PackageCompiler.create_sysimage(; sysimage_path="test_215.so", cpu_target)
target.name: znver3
sysimg.size(): 3
[0]: generic
[1]: sandybridge
[2]: haswell
Imaging:, 0
cmdline size, 1
CMDLINE: [0]: Flags: 0, Name: native
target.name: znver3
sysimg.size(): 3
[0]: generic
[1]: sandybridge
[2]: haswell
Imaging:, 0
cmdline size, 1
CMDLINE: [0]: Flags: 0, Name: native
target.name: znver3
sysimg.size(): 3
[0]: generic
[1]: sandybridge
[2]: haswell
Imaging:, 0
cmdline size, 1
CMDLINE: [0]: Flags: 0, Name: native
✔ [02m:40s] PackageCompiler: compiling incremental system image
```
</details>

**Comment by ven-k** [comment]
- https://github.com/JuliaLang/julia/pull/48809 with this we should be able to pass both `--cpu-target` and `--pkgimages` in the cmd. And sysimages build for all the targets.
- Not yet clear why setting `JULIA_CPU_TARGET` as an ENV var fails.

**Comment by BioTurboNick** [comment]
I'm not sure if this is relevant or if I'm encountering another issue - I'm trying to use PackageCompiler to create a Julia image for a Docker container, which will be deployed on an AWS computer. I waited to try on 1.9-rc3 because I saw the same thing on rc2 and saw this issue; but rc3 has the same problem. If there's something I can check to determine what the cause is, that would be helpful.

When run on AWS, it gives me `Unable to find compatible target in system image.`

Creating:
```
PackageCompiler.create_sysimage(packages;
    cpu_target = "generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)", # Julia default string
    sysimage_path = "julia-custom-sysimage.so",
    precompile_execution_file = "precompile.jl"
```

Running:
```
julia --threads auto --sysimage=julia-smlm-sysimage.so execute.jl
```
